src/components/remote_control/test/commands/set_interior_vehicle_data_request_test.cc

```diff
@@ -0,0 +1,411 @@
+  EXPECT_CALL(*mock_service_, GetApplication(mobile_message->connection_key()))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_service_, ValidateMessageBySchema(*mobile_message))
+      .WillOnce(Return(application_manager::MessageValidationResult::SUCCESS));
+  EXPECT_CALL(*mock_service_, CheckPolicyPermissions(mobile_message))
+      .WillOnce(Return(mobile_apis::Result::eType::SUCCESS));
+  application_manager::AppExtensionPtr invalid_ext;
+  EXPECT_CALL(*mock_app_, QueryInterface(kModuleId))
+      .WillOnce(Return(invalid_ext))
+      .WillRepeatedly(Return(rc_app_extention_));
+  application_manager::AppExtensionPtr app_extension;
+  EXPECT_CALL(*mock_app_, AddExtension(_))
+      .WillOnce(DoAll(SaveArg<0>(&app_extension), Return(true)));
+  EXPECT_CALL(*mock_service_, CheckAccess(_, _, _, _))
+      .WillOnce(Return(application_manager::TypeAccess::kAllowed));
+  EXPECT_CALL(*mock_service_, GetNextCorrelationID()).WillOnce(Return(1));
+  application_manager::MessagePtr result_msg;
+  EXPECT_CALL(*mock_service_, SendMessageToHMI(_))
+      .WillOnce(SaveArg<0>(&result_msg));
+  EXPECT_CALL(mock_allocation_manager_, AcquireResource("CLIMATE", app_id_))
+      .WillOnce(Return(remote_control::AcquireResult::ALLOWED));
+  // Act
+  remote_control::request_controller::MobileRequestPtr command =
+      CreateCommand(mobile_message);
+  command->Run();
+  // Assertions
+  EXPECT_EQ(kModuleId, app_extension->uid());
+  EXPECT_EQ(application_manager::ProtocolVersion::kHMI,
+            result_msg->protocol_version());
+  EXPECT_EQ(1, result_msg->correlation_id());
+  EXPECT_EQ(application_manager::MessageType::kRequest, result_msg->type());
+  const Json::Value hmi_request_params =
+      MessageHelper::StringToValue(result_msg->json_message());
+  EXPECT_EQ(functional_modules::hmi_api::set_interior_vehicle_data,
+            hmi_request_params[json_keys::kMethod].asString());
+  const Json::Value hmi_request_climate_control_data =
+      hmi_request_params[json_keys::kParams][message_params::kModuleData]
+                        [message_params::kClimateControlData];
+  EXPECT_EQ(2u, hmi_request_climate_control_data.size());
+  const Json::Value hmi_request_module_data_fan_speed =
+      hmi_request_climate_control_data[message_params::kFanSpeed];
+  const Json::Value hmi_request_module_data_defrost_zone =
+      hmi_request_climate_control_data[message_params::kDefrostZone];
+  EXPECT_EQ("FRONT", hmi_request_module_data_defrost_zone.asString());
+  EXPECT_EQ(10, hmi_request_module_data_fan_speed.asInt());
+}
+
+TEST_F(SetInteriorVehicleDataRequestTest,
+       Execute_ValidMessageWithoutSettableAndWithReadOnlyParams) {
+  // Arrange
+  application_manager::MessagePtr mobile_message = CreateBasicMessage();
+  mobile_message->set_json_message(
+      kValidMobileRequestWithoutSettableAndWithReadOnlyParams);
+  // Expectations
+  EXPECT_CALL(mock_module_, service()).WillOnce(Return(mock_service_));
```

Comment:
@Itileda you already created ON_CALL for service getter. expect is redundant. ( same as in prev UT)